### PR TITLE
Improve usability of the decryption banner

### PR DESCRIPTION
**Pull request by florianduros**
## Checklist

-   [x] Tests written for new code (and old code if feasible)
-   [x] Linter and other CI checks pass
-   [x] Sign-off given on the changes (see [CONTRIBUTING.md](https://github.com/matrix-org/matrix-react-sdk/blob/develop/CONTRIBUTING.md))

Closes https://github.com/vector-im/element-meta/issues/1358

- Replace `Decrypting messages…` description by a new one
- Reduce banner font
- Removed all the error messages where the user was stuck in a dead end. If the user can't decrypt the message, do not display permanently a banner about it. 
- Display an error message only when the device is unverified and an verified device or a key backup exist.
- Display a generic message when the `developerMode` is enabled, the device is verified, another device is verified and there is a unrequested session. It allows to send manually key request in `developerMode`.

---
Here's what your changelog entry will look like:

## ✨ Features
 * Improve usability of the decryption banner ([\#11012](https://github.com/matrix-org/matrix-react-sdk/pull/11012)). Fixes vector-im/element-meta#1358. Contributed by @florianduros.

### CODE REVIEW

File: src/components/views/rooms/DecryptionFailureBar.tsx [hunk 1]
```diff
@@ -154,78 +142,38 @@ export const DecryptionFailureBar: React.FC<IProps> = ({ failures }) => {
...
                 data-testid="decryption-failure-bar-button"
             >
-                {_t("View your device list")}
+                {_t("Verify")}
             </AccessibleButton>
         );
-    } else {
-        className = "mx_DecryptionFailureBar";
+        // In developerMode, we want to be able to resend manually key requests
+    } else if (developerMode && !needsVerification && hasOtherVerifiedDevices && anyUnrequestedSessions) {
```
Comment (richvdh) [inline src/components/views/rooms/DecryptionFailureBar.tsx:175]: The comment should really be inside the relevant `if` clause:

```suggestion
    } else if (developerMode && !needsVerification && hasOtherVerifiedDevices && anyUnrequestedSessions) {
        // In developerMode, show a button to resend key requests if there are other devices we could request keys from.
```

File: test/components/views/rooms/DecryptionFailureBar-test.tsx [hunk 2]
```diff
@@ -189,40 +190,12 @@ describe("<DecryptionFailureBar />", () => {
...
-        act(() => {
-            jest.advanceTimersByTime(5000);
-        });
-        expect(getBar(bar)).toMatchSnapshot();
-
-        bar.unmount();
-    });
-
-    it("Displays a general error message if there are no other verified devices", async () => {
+    it("The banner is hidden if there are no other verified devices", async () => {
```
Comment (richvdh) [inline test/components/views/rooms/DecryptionFailureBar-test.tsx:198]:
```suggestion
    it("The banner is hidden if there is a key backup but no other verified devices", async () => {
```

Comment (richvdh) [reply test/components/views/rooms/DecryptionFailureBar-test.tsx:198]: it seems like the code doesn't actually care about key backup or other verified devices?

**Review by richvdh — CHANGES_REQUESTED** [review]

**Comment by florianduros** [comment]
Irrelevant since https://github.com/vector-im/element-meta/issues/1358#issuecomment-1572049548

Closing it in favor of https://github.com/matrix-org/matrix-react-sdk/pull/11027.

**Pull request closed without merging**